src/components/SignName.vue

```diff
@@ -100,91 +100,108 @@ import axios from 'axios'
 
 export default {
 
-    name: 'SignName',
-    data: () => ({
-      errorMessages: '',
-      name: null,
-      line1: null,
-      line2: null,
-      city: null,
-      state: null,
-      zip: null,
-      country: null,
-      email: null,
-      formHasErrors: false,
-      JSONstring: '',
-      message: ''
-    }),
-
-    computed: {
-      form () {
-        return {
-          name: this.name,
-          line1: this.line1,
-          line2: this.line2,
-          city: this.city,
-          state: this.state,
-          zip: this.zip,
-          email: this.email
-        }
+  name: 'SignName',
+  data: () => ({
+    errorMessages: '',
+    name: null,
+    line1: null,
+    line2: null,
+    city: null,
+    state: null,
+    zip: null,
+    country: null,
+    email: null,
+    formHasErrors: false,
+    JSONstring: '',
+    message: ''
+  }),
+  isFormFilled: false,
+
+  computed: {
+    form() {
+      return {
+        name: this.name,
+        line1: this.line1,
+        line2: this.line2,
+        city: this.city,
+        state: this.state,
+        zip: this.zip,
+        email: this.email
       }
+    }
+  },
+
+  watch: {
+    name() {
+      this.errorMessages = ''
     },
+    form: {
```

Comment:
Do tell me what you think, I would to know if  this is a right implimentation of a watcher in vue